### DISCUSSION TITLE
Glicol and Bevy

### DISCUSSION THREAD

**Discussion by harudagondi** [chaosprint/glicol#112] [General]
Hello! I am interested in integrating a bunch of audio libraries into [Bevy], a Rust game engine. Currently I have written a library for integrating [FunDSP] into Bevy (it's called [`bevy_fundsp`]). However, I found out recently that fundsp isn't really viable for exposing its API into a scripting language (like rhai). However, I recently discovered glicol, and I would like to use this in game development.

Is there a reason why glicol isn't published yet in crates.io?

[Bevy]: https://bevyengine.org/
[FunDSP]: https://github.com/SamiPerttu/fundsp
[`bevy_fundsp`]: https://github.com/harudagondi/bevy_fundsp

**Comment by chaosprint** [comment]
Hi,

Thanks for the post! The integration looks interesting. The reason why Glicol is not on crates.io yet is that I am working on solving the channel matching issue. Without confirming these structural mechanisms, it would be very troublesome to publish now and refactor soon in the future. There are some other syntax issues to consider as well so the `glicol_synth` can be refactored drastically accordingly. 

Once this is done, adding new nodes is fairly easy.

Unfortunately, I have quite a lot of personal issues to solve at the moment. So I can only work towards it gradually but I think once I am fully back to coding again it will be there soon.

**Reply by harudagondi** [reply]
Is there a Github issue for the channel matching?

Also, no worries. This isn't urgent :)

**Reply by chaosprint** [reply]
I guess it's this one #28 

The limitation for two channels only is embodied with some hard-coded stuff. And I need to figure out what mechanism should be used when connecting a mono node to a stereo node, as well as the opposite.

**Comment by orottier** [comment]
Hey @chaosprint, if you wish I could prepare a 0.1 release of the components and transfer the ownership to you on crates.io
It would be sad if someone would namesquat the glicol name and use it for something else.
All 0.x versions are considerd major version releases by cargo so you should not worry about future refactors!

**Reply by harudagondi** [reply]
I think it's better if it is a 0.0.1 version first, so chaosprint can make a 0.1 release if they wish to do so.

**Reply by chaosprint** [reply]
Hi, that's a good point and thanks for offering the help. I can manage to publish those crates today, although the docs are not there yet. I'd like to keep them in sync with the current repo tag.

**Reply by chaosprint** [reply]
It's there now. 
https://crates.io/search?q=glicol
I think the `glicol_synth` should be the one that is used most frequently.

**Reply by orottier** [reply]
Great!

**Comment by chaosprint** [comment]
https://github.com/chaosprint/bevy_glicol
